packages/playground/css/postcss.config.js

```diff
@@ -16,18 +16,25 @@ function testDirDep() {
     AtRule(atRule, { result, Comment }) {
       if (atRule.name === 'test') {
         const pattern = normalizePath(
-          path.resolve(path.dirname(result.opts.from), './glob-dep/*.css')
+          path.resolve(path.dirname(result.opts.from), './glob-dep/*.{css,js}')
         )
         const files = glob.sync(pattern)
-        const text = files.map((f) => fs.readFileSync(f, 'utf-8')).join('\n')
+        const text = files
+          .map((f) => {
+            if (f.endsWith('.css')) {
+              return fs.readFileSync(f, 'utf-8')
+            }
+            return require(f)
```

Comment:
The test case fails because `require` calls are cached. If you add `delete require.cache[f]` before requiring then it works as expected.

Reply:
Ah! Adding `delete require.cache[f]` doesn't fix it for me, but changing it to `eval(fs.readFileSync(f, 'utf-8'))` does 👍

In that case do you have any idea why the example project doesn't work? https://github.com/bradlc/vite-postcss-dependencies

Reply:
(I can open a separate issue if that would be useful?)

Reply:
Looked into it a bit and found out it is indeed a bug - only happens in the demo because editing `main.js` also happens to trigger a full page reload - and causing the code path that invalidates the CSS to be skipped.

Fixed in e048114d

Reply:
I have updated the issue and reproduction here: https://github.com/vitejs/vite/issues/3717

It seems like the same issue (or a very similar one) is still present for `.vue` files.